src/Microsoft.IdentityModel.Tokens/Encryption/AuthenticatedEncryptionProvider.cs

```diff
@@ -65,18 +65,29 @@ public AuthenticatedEncryptionProvider(SecurityKey key, string algorithm)
             if (string.IsNullOrWhiteSpace(algorithm))
                 throw LogHelper.LogArgumentNullException(nameof(algorithm));
 
-            if (!IsSupportedAlgorithm(key, algorithm))
-                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX10668, GetType(), algorithm, key)));
+            // Allow constructing AuthenticatedEncryptionProvider in case when AesGcm algorithm is passed to enable tests
+            // Until there is no AES-GCM support, Encrypt and Decrypt operations will throw an exception
+            if (IsAesGcmAlgorithm(algorithm))
+            {
+                Key = key;
+                Algorithm = algorithm;
+            }
+            else
+            {
+                if (!IsSupportedAlgorithm(key, algorithm))
+                    throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX10668, GetType(), algorithm, key)));
 
-            ValidateKeySize(key, algorithm);
-            _authenticatedkeys = GetAlgorithmParameters(key, algorithm);
-            _hmacAlgorithm = GetHmacAlgorithm(algorithm);
-            _symmetricSignatureProvider = key.CryptoProviderFactory.CreateForSigning(_authenticatedkeys.HmacKey, _hmacAlgorithm) as SymmetricSignatureProvider;
-            if (_symmetricSignatureProvider == null)
-                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX10649, Algorithm)));
+                ValidateKeySize(key, algorithm);
 
-            Key = key;
-            Algorithm = algorithm;
+                _authenticatedkeys = GetAlgorithmParameters(key, algorithm);
+                _hmacAlgorithm = GetHmacAlgorithm(algorithm);
+                _symmetricSignatureProvider = key.CryptoProviderFactory.CreateForSigning(_authenticatedkeys.HmacKey, _hmacAlgorithm) as SymmetricSignatureProvider;
+                if (_symmetricSignatureProvider == null)
+                    throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX10649, Algorithm)));
+
+                Key = key;
+                Algorithm = algorithm;
```

Comment:
suggestion: move setting key/algorithm to after if/else. This means you can make the above just an if (!isAes...)
objective: reduce nesting for readability.

Reply:
I'll put a comment on top of AES-GCM algo definitions to improve clarity and set expectations.
Also, I'll refactor the constructor to increase current readability. 
When we add AES-GCM support it might be better to return to *if (isAesGcm)* to separate support for AES-GCM from other authenticated encryption algorithms.

Reply:
addressed with: c9daa02e155a336edf2c4c8cfff624ccf82277a1 and c80fbb9ae2e1d68898fe6843df0881e1a315e3a2